contracts/game.sol

```diff
@@ -0,0 +1,125 @@
+    _;
+  }
+
+  function isJudge(address judge) view public returns (bool) {
+    bool contains = false;
+    for (uint i = 0; i < judges.length; i++) {
+      if (judge == judges[i]) {
+        contains = true;
+      }
+    }
+    return contains;
+  }
+
+  function hasTokens(address player) view public returns (bool) {
+    return _token.balanceOf(player) >= 100 * 10**18;
+  }
+
+  function hasTooManyTokens(address player) view public returns (bool) {
+    return _token.balanceOf(player) >= 1000 * 10**18;
+  }
+
+  function hasBeenVerifiedToWin(address player) view public returns (bool) {
+    bool contains = false;
+    for (uint i = 0; i < playersVerifiedToWin.length; i++) {
+      if (player == playersVerifiedToWin[i]) {
+        contains = true;
+      }
+    }
+    return contains;
+  }
+
+  function hasNotWonBefore(address player) view public returns (bool) {
+    return !hasWon(player);
+  }
+
+  function seekVerification() public {
+    require(hasTokens(msg.sender), "Need 100 FWEB3 tokens");
+    require(!hasTooManyTokens(msg.sender), "Need less than 1,000 FWEB3 tokens");
+    playersSeekingVerification.push(payable(msg.sender));
+    emit PlayerSeeksVerification(msg.sender);
+  }
+
+  function win() public {
+    require(hasTokens(msg.sender), "Need 100 FWEB3 tokens");
+    require(!hasTooManyTokens(msg.sender), "Need less than 1,000 FWEB3 tokens");
+    require(hasBeenVerifiedToWin(msg.sender), "Need to have been verified by a judge");
+    require(hasNotWonBefore(msg.sender), "Need to have not won before");
+    (bool sent, ) = (msg.sender).call{ value: 1000 * (10*18) }("");
+    require(sent, "Failed to send Ether");
+    winners.push(payable(msg.sender));
+    emit PlayerWon(msg.sender);
+  }
+
+  function approveWinner(address player) public judgeOnly {
+    removePlayerFromSeekingVerification(player);
+    playersVerifiedToWin.push(payable(player));
+    emit PlayerVerifiedToWin(player, msg.sender);
+  }
+
+  function rejectWinner(address player) public {
+    removePlayerFromSeekingVerification(player);
+  }
+
+  function addJudge(address judge) public onlyOwner {
+    require(!isJudge(judge), "Already a judge");
+    judges.push(payable(judge));
+  }
+
+  function removeJudge(address judge) public onlyOwner {
+    for (uint i = 0; i < judges.length; i++) {
+      if (judge == judges[i]) {
+        delete judges[i];
+      }
+    }
+  }
+
+  function getPlayersSeekingVerification() view public returns (address payable[] memory) {
+    return playersSeekingVerification;
+  }
+
+  function getNumWinners() view public returns (uint) {
+    return winners.length;
+  }
+
+  function hasWon(address player) view public returns (bool) {
+    bool contains = false;
+    for (uint i = 0; i < winners.length; i++) {
+      if (player == winners[i]) {
+        contains = true;
+      }
+    }
+    return contains;
+  }
+
+  function removePlayerFromSeekingVerification(address player) internal {
```

Comment:
Same comment here. With a long enough array, this will become very expensive quickly. Mappings would be a better way to do this to keep the cost down and keep the same structure.